adapter/internal/notifier/deployment_notifier.go

```diff
@@ -82,3 +87,52 @@ func SendRevisionUpdate(deployedRevisionList []*DeployedAPIRevision) {
 		}
 	}
 }
+
+// SendRevisionUndeploy - send the undeployed revision information to control plane
+func SendRevisionUndeploy(apiUUID string, revisionUUID string, environment string) {
+	conf, _ := config.ReadConfigs()
+	cpConfigs := conf.ControlPlane
+
+	revisionEP := cpConfigs.ServiceURL
+	if strings.HasSuffix(revisionEP, "/") {
+		revisionEP += unDeployedRevisionEP
+	} else {
+		revisionEP += "/" + unDeployedRevisionEP
+	}
+
+	if apiUUID == "" || revisionUUID == "" || environment == "" {
```

Comment:
Don't we have to check if the control plane is enabled here?

Reply:
Fixed via : https://github.com/wso2/product-microgateway/pull/2503/commits/4d77d188a4aefb79e5d8582c06613d46ae0a1348